In [ ]:
from ultralytics import YOLO

In [3]:
# Load model
model = YOLO("models/detection/yolo11x_finetuned_bottles_on_site_v3_copy.pt")

In [5]:
# Explore this model
model.info()

YOLO11x summary: 357 layers, 56,874,931 parameters, 0 gradients, 195.4 GFLOPs


(357, 56874931, 0, 195.44908800000002)

| Field | Meaning |
|:-------|:---------|
| **357 layers** | Depth (complexity) of the YOLO11x model architecture |
| **56.9M parameters** | Total number of learnable weights (model size) |
| **0 gradients** | Indicates the model isn’t currently in training mode |
| **195.4 GFLOPs** | Approximate computational cost per image (billions of operations) |

In [7]:
# Check performance of this model
results = model.val(data="explore_model_val.yaml")

Ultralytics 8.3.205 🚀 Python-3.11.5 torch-2.2.2 CPU (Intel Core i5-6287U 3.10GHz)
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 215.2±114.3 MB/s, size: 1757.1 KB)
val: Scanning /Users/chris/Desktop/UCLL 2025-2026/AI Applications/play-around/images/detection/fine_tune_yolo/bottles_on_site/labels/val... 21 images, 1 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 22/22 177.9it/s 0.1s.0s
val: New cache created: /Users/chris/Desktop/UCLL 2025-2026/AI Applications/play-around/images/detection/fine_tune_yolo/bottles_on_site/labels/val.cache
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 2/2 0.1it/s 34.9s1:26
                   all         22         28      0.957      0.964      0.992      0.901
Speed: 6.3ms preprocess, 1487.5ms inference, 0.0ms loss, 1.0ms postprocess per image
Results saved to /Users/chris/Desktop/UCLL 2025-2026/AI Applications/play-around/runs/detect/val2


 !!! You don't need to specify the label path in the evaluation YAML because the framework is designed to infer it from the image path, provided your dataset is structured correctly.